supertokens_python/recipe/passwordless/utils.py

```diff
@@ -184,37 +173,49 @@ def __init__(self,
                  override: OverrideConfig,
                  flow_type: Literal['USER_INPUT_CODE', 'MAGIC_LINK', 'USER_INPUT_CODE_AND_MAGIC_LINK'],
                  get_link_domain_and_path: Callable[[PhoneOrEmailInput, Dict[str, Any]], Awaitable[str]],
+                 get_email_delivery_config: Callable[[], EmailDeliveryConfigWithService[TypePasswordlessEmailDeliveryInput]],
                  get_custom_user_input_code: Union[Callable[[
-                     Dict[str, Any]], Awaitable[str]], None] = None
+                     Dict[str, Any]], Awaitable[str]], None] = None,
                  ):
         self.contact_config = contact_config
         self.override = override
         self.flow_type: Literal['USER_INPUT_CODE', 'MAGIC_LINK', 'USER_INPUT_CODE_AND_MAGIC_LINK'] = flow_type
         self.get_custom_user_input_code = get_custom_user_input_code
         self.get_link_domain_and_path = get_link_domain_and_path
+        self.get_email_delivery_config = get_email_delivery_config
 
 
 def validate_and_normalise_user_input(
-        app_info: AppInfo,
-        contact_config: ContactConfig,
-        flow_type: Literal['USER_INPUT_CODE', 'MAGIC_LINK', 'USER_INPUT_CODE_AND_MAGIC_LINK'],
-        override: Union[OverrideConfig, None] = None,
-        get_link_domain_and_path: Union[Callable[[
-            PhoneOrEmailInput, Dict[str, Any]], Awaitable[str]], None] = None,
-        get_custom_user_input_code: Union[Callable[[Dict[str, Any]], Awaitable[str]], None] = None):
+    app_info: AppInfo,
+    contact_config: ContactConfig,
+    flow_type: Literal['USER_INPUT_CODE', 'MAGIC_LINK', 'USER_INPUT_CODE_AND_MAGIC_LINK'],
+    override: Union[OverrideConfig, None] = None,
+    get_link_domain_and_path: Union[Callable[[
+        PhoneOrEmailInput, Dict[str, Any]], Awaitable[str]], None] = None,
+    get_custom_user_input_code: Union[Callable[[Dict[str, Any]], Awaitable[str]], None] = None,
+    email_delivery: Union[EmailDeliveryConfig[TypePasswordlessEmailDeliveryInput], None] = None,
+) -> PasswordlessConfig:
 
     if override is None:
         override = OverrideConfig()
 
     if get_link_domain_and_path is None:
         get_link_domain_and_path = default_get_link_domain_and_path(app_info)
 
+    def get_email_delivery_config() -> EmailDeliveryConfigWithService[TypePasswordlessEmailDeliveryInput]:
+        email_service = email_delivery.service if email_delivery is not None else None
+        if email_service is None:
+            email_service = BackwardCompatibilityService(app_info)
+
+        return EmailDeliveryConfigWithService(email_service, override=None)
```

Comment:
This is not taking into account any of the `create_and_send_custom_email`.

Reply:
Fixed.